src/Symfony/Component/Form/Extension/Core/Type/TextType.php

```diff
@@ -33,4 +47,22 @@ public function getBlockPrefix()
     {
         return 'text';
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($data)
+    {
+        // Model data should not be transformed
+        return $data;
+    }
+
+    /**
+     * {@inheritdoc}
+     *.
+     */
+    public function reverseTransform($data)
+    {
+        return empty($data) ? '' : $data;
```

Comment:
@fsevestre Thanks! I've changed it will making the tests and forgot to revert it :)